clients/src/main/java/org/apache/kafka/common/requests/PushTelemetryRequest.java

```diff
@@ -83,8 +94,38 @@ public Uuid getClientInstanceId() {
         return this.data.clientInstanceId();
     }
 
+    public int getSubscriptionId() {
+        return this.data.subscriptionId();
+    }
+
+    public boolean isClientTerminating() {
+        return this.data.terminating();
+    }
+
+    public String getMetricsContentType() {
+        // Future versions of PushTelemetryRequest and GetTelemetrySubscriptionsRequest may include a content-type
+        // field to allow for updated OTLP format versions (or additional formats), but this field is currently not
+        // included since only one format is specified in the current proposal of the kip-714
+        return "OTLP";
+    }
+
+    public ByteBuffer getMetricsData() throws Exception {
+        CompressionType cType = CompressionType.forId(this.data.compressionType());
```

Comment:
Not sure if this handles the case OK, but make sure that a bogus value for the compression type from the client is handled well.

Reply:
There is a validation check while in processing pushTelemetryRequest where we validate all the parameters in the message. Here we can safely assume that we passed all those checks.